src/json/next/serialization.cr

```diff
@@ -0,0 +1,217 @@
+module JSON
+  annotation Field
+  end
+
+  module Serializable
+    annotation Options
+    end
+
+    macro included
+      def self.new(pull : ::JSON::PullParser)
```

Comment:
Why is this required at all?

Reply:
Try it for yourself. If you remove it, it stops working. But I have no idea why. It's just a hack, but it works.

Reply:
as i understand `new` calls always before/instead of `initialize` with the same params.